site/components/Playground.tsx

```diff
@@ -53,23 +62,102 @@ export function Playground() {
       result: output.result(),
       result_hash: output.result_hash(),
       hashes: output.hashes(),
-      // selfDestructed: output.self_destructed(),
+      selfDestructed: output.self_destructed(),
       readAuth: output.read_auth(),
     }, null, 2))
   })
 
-  const verify = useAsyncCallback(() => {
-    if (!output) {
+  // Uses the /prove endpoint
+  const prove_server = useAsyncCallback(async () => {
+    clearOutput.execute()
+
+    const proverUrl = 'http://127.0.0.1:8080/prove'
+    const parsedInputs = JSON.parse(inputs)
+
+    const { midenCode, abi } = compile(code, parsedInputs)
+
+    const payload = {
+      midenCode: midenCode,
+      abi: abi,
+      this: parsedInputs.init_params,
+      args: parsedInputs.params,
+    }
+
+    fetch(proverUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(payload),
+    }).then(resp => resp.json())
```

Comment:
Try to avoid using `.then` with promises, you can just await the result - 

```js
const resp = await fetch(...)
const json = await resp.json()
````

etc